### PR TITLE
Feature/create services

### PR DESCRIPTION
**Pull request by Dz0n1Pr0**

### CODE REVIEW

File: Backend/src/main/java/xmlteam4/Project/controllers/CoverLetterController.java [hunk 1]
```diff
@@ -1,22 +1,41 @@
...
     @GetMapping(value = "/{id}", produces = MediaType.TEXT_HTML_VALUE)
     public ResponseEntity<String> getCoverLetter(@PathVariable("id") String id) {
-        throw new NotImplementedException();
+        String cover = "";
+        try {
+            cover = coverLetterService.findOne(id);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return new ResponseEntity<>(cover, HttpStatus.OK);
```
Comment (draganculibrk9) [inline Backend/src/main/java/xmlteam4/Project/controllers/CoverLetterController.java]: Method should be:

`
try {
    return new ResponseEntity<>(coverLetterService.findOne(id), HttpStatus.OK);
} catch (Exception e) {
    return new ResponseEntity<>("Failed to find cover letter with given id", HttpStatus.NOT_FOUND)
}
`
Same goes for others. Use appropriate status codes!

File: Backend/src/main/java/xmlteam4/Project/services/ScientificPaperService.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+        return paper;
+
+    }
+
+    public String create(String xml) throws Exception {
+        Document document = domParser.buildDocument(xml, scientificPaperSchemaPath);
+
+        String id = IDGenerator.createID();
+        document.getDocumentElement().setAttribute("id", id);
```
Comment (draganculibrk9) [inline Backend/src/main/java/xmlteam4/Project/services/ScientificPaperService.java]: After generating an ID for scientific paper, you should generate id for every child element which can have an ID(read schema for more info). ID should be URL and generation can be done using sequencer. Check other document types for the same issue.

File: Backend/src/main/java/xmlteam4/Project/services/ScientificPaperService.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+        String newXml = documentXMLTransformer.toXMLString(document);
```
Comment (draganculibrk9) [inline Backend/src/main/java/xmlteam4/Project/services/ScientificPaperService.java:56]: Metadata is not set. Read schema and project specification for more constraints.

**Review by draganculibrk9 — CHANGES_REQUESTED** [review]

**Review by draganculibrk9 — CHANGES_REQUESTED** [review]
Some constraints are not met yet. Check workflow.

File: Backend/src/main/java/xmlteam4/Project/controllers/CoverLetterController.java (hunk 1 above)
Comment (Dz0n1Pr0) [reply Backend/src/main/java/xmlteam4/Project/controllers/CoverLetterController.java]: Methods changed accordingly

File: Backend/src/main/java/xmlteam4/Project/services/ScientificPaperService.java (hunk 2 above)
Comment (Dz0n1Pr0) [reply Backend/src/main/java/xmlteam4/Project/services/ScientificPaperService.java]: ID's have been generated for all child elements and ID is now an URL

File: Backend/src/main/java/xmlteam4/Project/services/ScientificPaperService.java (hunk 3 above)
Comment (Dz0n1Pr0) [reply Backend/src/main/java/xmlteam4/Project/services/ScientificPaperService.java:56]: Metadata is now set

**Review by draganculibrk9 — APPROVED** [review]

**Pull request merged**